src/website/src/app/documentation/demos/datalist/datalist.demo.html

```diff
@@ -0,0 +1,92 @@
+<!--
+  ~ Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<clr-doc-wrapper [ng]="ng" [ui]="ui" [title]="title" [newLayout]="true">
+    <article>
+        <h5 class="component-summary">
+            The datalist element offers a flexible input when users need to filter and select from a large
+            list of pre-defined options. Or, they need to input a custom value (not provided in the pre-defined list)
+            for the input.
+        </h5>
+
+        <div class="design-guidelines">
+            <h3>Usage</h3>
+
+            <p>
+                Use a datalist when the user needs to choose from a list of items that is long enough for it to be
+                filtered it before selection. Or, there are pre-defined options but the user needs the ability to input
+                a custom value for their use case.
+            </p>
+
+            <p>
+                Use a datalist of you want to provide a standard set of options but also allow the user to enter a
+                custom value that may not be pre-defined.
+            </p>
+
+            <p>
+                Do not confuse a datalist element with the select input. Datalist elements usually have more than 13
+                options that the user will need to filter or narrow down before making a choice. If there are 3-13
+                options, consider using the <a routerLink="../../select">select</a> input.
+            </p>
+
+            <h3>Auto-completion</h3>
+
+            <p>
+                Autocomplete is a familiar pattern for users of web technology. Searching, item suggestion and
+                filtering are all common activities users encounter when interacting with web technology and content.
+                The HTML5 element <code class="clr-code">dataliste</code> natively supports autocomplete behavior on
```

Comment:
there is a typo `datalist`

Reply:
This seems to be related to differences between vertical/horizontal form layouts. I'm looking into it.

Reply:
This happens in horizontal layout b/c some inputs need  the right margin for an icon (datepicker). Vertical layouts remove the right margin. 
I added a targeted select for `clr-input-group` 's that are inside of the `clr-datalist-container`.

Reply:
I have almost the same issue with the datepicker additional clear icon, Maybe we need to rework this icon and input ordering pattern. 

I will open the conversation thread cause the typo is still there
```html
<code class="clr-code">dataliste</code>
```